packages/type-utils/src/isTypeReadonly.ts

```diff
@@ -107,9 +107,18 @@ function isTypeReadonlyObject(
   function checkIndexSignature(kind: ts.IndexKind): Readonlyness {
     const indexInfo = checker.getIndexInfoOfType(type, kind);
     if (indexInfo) {
-      return indexInfo.isReadonly
-        ? Readonlyness.Readonly
-        : Readonlyness.Mutable;
+      if (!indexInfo.isReadonly) {
+        return Readonlyness.Mutable;
+      }
+
+      return (
+        isTypeReadonlyRecurser(
+          checker,
+          indexInfo.keyType,
```

Comment:
question: do we care if the key type is mutable?
The key type will always be a `number | string | symbol` - so they are all technically immutable.
(or am I misunderstanding this code?)

Reply:
Yeah, that's true. I'll remove that